### ISSUE TITLE
[rllib] Should auto restrict GPU usage by setting CUDA_VISIBLE_DEVICES

### ISSUE THREAD

**Issue by ericl** [ray-project/ray#975]
Currently, applications have to manually specify which devices are visible to e.g. each actor. The runtime should probably set these restrictions by default so the user does not need to (and in general, when actors are scheduled across multiple machines, it is difficult to do this assignment manually).

```
@ray.remote(num_gpus=2)
class Foo:
   # in this worker CUDA_VISIBLE_DEVICES=$A,$B where A and B
   # are chosen by Ray from available GPUs.
``

**Comment by robertnishihara** [comment]
Do you mean we should run the following under the hood whenever we start a GPU task or GPU actor?

```python
os.environ['CUDA_VISIBLE_DEVICES'] = ','.join([str(i) for i in ray.get_gpu_ids()])
```

That seems reasonable.

**Comment by robertnishihara** [comment]
Fixed by #1044.
